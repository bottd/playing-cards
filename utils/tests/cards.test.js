const cards = require('../cards');

describe('Cards', () => {
  describe('Deck', () => {
    it('Should have a 52 cards', () => {
      expect(cards.deck.length).toBe(52);
    });
    it('Should match snapshot', () => {
      expect(cards.deck).toMatchSnapshot();
    });
  });

  describe('Joker', () => {
    it('Should match snapshot', () => {
      expect(cards.joker).toMatchSnapshot();
    });
  });

  describe('Deck with Jokers', () => {
    it('Should have 54 cards', () => {
      expect(cards.deckWithJokers.length).toBe(54);
    });
    it('Should match snapshot', () => {
      expect(cards.deckWithJokers).toMatchSnapshot();
    });
  });
});
