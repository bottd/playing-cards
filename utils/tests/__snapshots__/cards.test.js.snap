// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards Deck Should match snapshot 1`] = `
Array [
  Object {
    "hidden": true,
    "id": 1,
    "name": "Ace of Spades",
    "suite": "spades",
    "symbol": "ace",
    "value": "1",
  },
  Object {
    "hidden": true,
    "id": 2,
    "name": "Two of Spades",
    "suite": "spades",
    "symbol": "two",
    "value": "2",
  },
  Object {
    "hidden": true,
    "id": 3,
    "name": "Three of Spades",
    "suite": "spades",
    "symbol": "three",
    "value": "3",
  },
  Object {
    "hidden": true,
    "id": 4,
    "name": "Four of Spades",
    "suite": "spades",
    "symbol": "four",
    "value": "4",
  },
  Object {
    "hidden": true,
    "id": 5,
    "name": "Five of Spades",
    "suite": "spades",
    "symbol": "five",
    "value": "5",
  },
  Object {
    "hidden": true,
    "id": 6,
    "name": "Six of Spades",
    "suite": "spades",
    "symbol": "six",
    "value": "6",
  },
  Object {
    "hidden": true,
    "id": 7,
    "name": "Seven of Spades",
    "suite": "spades",
    "symbol": "seven",
    "value": "7",
  },
  Object {
    "hidden": true,
    "id": 8,
    "name": "Eight of Spades",
    "suite": "spades",
    "symbol": "eight",
    "value": "8",
  },
  Object {
    "hidden": true,
    "id": 9,
    "name": "Nine of Spades",
    "suite": "spades",
    "symbol": "nine",
    "value": "9",
  },
  Object {
    "hidden": true,
    "id": 10,
    "name": "10 of Spades",
    "suite": "spades",
    "symbol": "ten",
    "value": "10",
  },
  Object {
    "hidden": true,
    "id": 11,
    "name": "Jack of Spades",
    "suite": "spades",
    "symbol": "jack",
    "value": "11",
  },
  Object {
    "hidden": true,
    "id": 12,
    "name": "Queen of Spades",
    "suite": "spades",
    "symbol": "queen",
    "value": "12",
  },
  Object {
    "hidden": true,
    "id": 13,
    "name": "King of Spades",
    "suite": "spades",
    "symbol": "king",
    "value": "13",
  },
  Object {
    "hidden": true,
    "id": 14,
    "name": "Ace of Clubs",
    "suite": "clubs",
    "symbol": "ace",
    "value": "1",
  },
  Object {
    "hidden": true,
    "id": 15,
    "name": "Two of Clubs",
    "suite": "clubs",
    "symbol": "two",
    "value": "2",
  },
  Object {
    "hidden": true,
    "id": 16,
    "name": "Three of Clubs",
    "suite": "clubs",
    "symbol": "three",
    "value": "3",
  },
  Object {
    "hidden": true,
    "id": 17,
    "name": "Four of Clubs",
    "suite": "clubs",
    "symbol": "four",
    "value": "4",
  },
  Object {
    "hidden": true,
    "id": 18,
    "name": "Five of Clubs",
    "suite": "clubs",
    "symbol": "five",
    "value": "5",
  },
  Object {
    "hidden": true,
    "id": 19,
    "name": "Six of Clubs",
    "suite": "clubs",
    "symbol": "six",
    "value": "6",
  },
  Object {
    "hidden": true,
    "id": 20,
    "name": "Seven of Clubs",
    "suite": "clubs",
    "symbol": "seven",
    "value": "7",
  },
  Object {
    "hidden": true,
    "id": 21,
    "name": "Eight of Clubs",
    "suite": "clubs",
    "symbol": "eight",
    "value": "8",
  },
  Object {
    "hidden": true,
    "id": 22,
    "name": "Nine of Clubs",
    "suite": "clubs",
    "symbol": "nine",
    "value": "9",
  },
  Object {
    "hidden": true,
    "id": 23,
    "name": "10 of Clubs",
    "suite": "clubs",
    "symbol": "ten",
    "value": "10",
  },
  Object {
    "hidden": true,
    "id": 24,
    "name": "Jack of Clubs",
    "suite": "clubs",
    "symbol": "jack",
    "value": "11",
  },
  Object {
    "hidden": true,
    "id": 25,
    "name": "Queen of Clubs",
    "suite": "clubs",
    "symbol": "queen",
    "value": "12",
  },
  Object {
    "hidden": true,
    "id": 26,
    "name": "King of Clubs",
    "suite": "clubs",
    "symbol": "king",
    "value": "13",
  },
  Object {
    "hidden": true,
    "id": 27,
    "name": "Ace of Diamonds",
    "suite": "diamonds",
    "symbol": "ace",
    "value": "1",
  },
  Object {
    "hidden": true,
    "id": 28,
    "name": "Two of Diamonds",
    "suite": "diamonds",
    "symbol": "two",
    "value": "2",
  },
  Object {
    "hidden": true,
    "id": 29,
    "name": "Three of Diamonds",
    "suite": "diamonds",
    "symbol": "three",
    "value": "3",
  },
  Object {
    "hidden": true,
    "id": 30,
    "name": "Four of Diamonds",
    "suite": "diamonds",
    "symbol": "four",
    "value": "4",
  },
  Object {
    "hidden": true,
    "id": 31,
    "name": "Five of Diamonds",
    "suite": "diamonds",
    "symbol": "five",
    "value": "5",
  },
  Object {
    "hidden": true,
    "id": 32,
    "name": "Six of Diamonds",
    "suite": "diamonds",
    "symbol": "six",
    "value": "6",
  },
  Object {
    "hidden": true,
    "id": 33,
    "name": "Seven of Diamonds",
    "suite": "spades",
    "symbol": "seven",
    "value": "7",
  },
  Object {
    "hidden": true,
    "id": 34,
    "name": "Eight of Diamonds",
    "suite": "diamonds",
    "symbol": "eight",
    "value": "8",
  },
  Object {
    "hidden": true,
    "id": 35,
    "name": "Nine of Diamonds",
    "suite": "diamonds",
    "symbol": "nine",
    "value": "9",
  },
  Object {
    "hidden": true,
    "id": 36,
    "name": "10 of Diamonds",
    "suite": "diamonds",
    "symbol": "ten",
    "value": "10",
  },
  Object {
    "hidden": true,
    "id": 37,
    "name": "Jack of Diamonds",
    "suite": "diamonds",
    "symbol": "jack",
    "value": "11",
  },
  Object {
    "hidden": true,
    "id": 38,
    "name": "Queen of Diamonds",
    "suite": "diamonds",
    "symbol": "queen",
    "value": "12",
  },
  Object {
    "hidden": true,
    "id": 39,
    "name": "King of Diamonds",
    "suite": "diamonds",
    "symbol": "king",
    "value": "13",
  },
  Object {
    "hidden": true,
    "id": 40,
    "name": "Ace of Hearts",
    "suite": "hearts",
    "symbol": "ace",
    "value": "1",
  },
  Object {
    "hidden": true,
    "id": 41,
    "name": "Two of Hearts",
    "suite": "hearts",
    "symbol": "two",
    "value": "2",
  },
  Object {
    "hidden": true,
    "id": 42,
    "name": "Three of Hearts",
    "suite": "hearts",
    "symbol": "three",
    "value": "3",
  },
  Object {
    "hidden": true,
    "id": 43,
    "name": "Four of Hearts",
    "suite": "hearts",
    "symbol": "four",
    "value": "4",
  },
  Object {
    "hidden": true,
    "id": 44,
    "name": "Five of Hearts",
    "suite": "hearts",
    "symbol": "five",
    "value": "5",
  },
  Object {
    "hidden": true,
    "id": 45,
    "name": "Six of Hearts",
    "suite": "hearts",
    "symbol": "six",
    "value": "6",
  },
  Object {
    "hidden": true,
    "id": 46,
    "name": "Seven of Hearts",
    "suite": "hearts",
    "symbol": "seven",
    "value": "7",
  },
  Object {
    "hidden": true,
    "id": 47,
    "name": "Eight of Hearts",
    "suite": "hearts",
    "symbol": "eight",
    "value": "8",
  },
  Object {
    "hidden": true,
    "id": 48,
    "name": "Nine of Hearts",
    "suite": "hearts",
    "symbol": "nine",
    "value": "9",
  },
  Object {
    "hidden": true,
    "id": 49,
    "name": "10 of Hearts",
    "suite": "hearts",
    "symbol": "ten",
    "value": "10",
  },
  Object {
    "hidden": true,
    "id": 50,
    "name": "Jack of Hearts",
    "suite": "hearts",
    "symbol": "jack",
    "value": "11",
  },
  Object {
    "hidden": true,
    "id": 51,
    "name": "Queen of Hearts",
    "suite": "hearts",
    "symbol": "queen",
    "value": "12",
  },
  Object {
    "hidden": true,
    "id": 52,
    "name": "King of Hearts",
    "suite": "hearts",
    "symbol": "king",
    "value": "13",
  },
]
`;

exports[`Cards Deck with Jokers Should match snapshot 1`] = `
Array [
  Object {
    "hidden": true,
    "id": 1,
    "name": "Ace of Spades",
    "suite": "spades",
    "symbol": "ace",
    "value": "1",
  },
  Object {
    "hidden": true,
    "id": 2,
    "name": "Two of Spades",
    "suite": "spades",
    "symbol": "two",
    "value": "2",
  },
  Object {
    "hidden": true,
    "id": 3,
    "name": "Three of Spades",
    "suite": "spades",
    "symbol": "three",
    "value": "3",
  },
  Object {
    "hidden": true,
    "id": 4,
    "name": "Four of Spades",
    "suite": "spades",
    "symbol": "four",
    "value": "4",
  },
  Object {
    "hidden": true,
    "id": 5,
    "name": "Five of Spades",
    "suite": "spades",
    "symbol": "five",
    "value": "5",
  },
  Object {
    "hidden": true,
    "id": 6,
    "name": "Six of Spades",
    "suite": "spades",
    "symbol": "six",
    "value": "6",
  },
  Object {
    "hidden": true,
    "id": 7,
    "name": "Seven of Spades",
    "suite": "spades",
    "symbol": "seven",
    "value": "7",
  },
  Object {
    "hidden": true,
    "id": 8,
    "name": "Eight of Spades",
    "suite": "spades",
    "symbol": "eight",
    "value": "8",
  },
  Object {
    "hidden": true,
    "id": 9,
    "name": "Nine of Spades",
    "suite": "spades",
    "symbol": "nine",
    "value": "9",
  },
  Object {
    "hidden": true,
    "id": 10,
    "name": "10 of Spades",
    "suite": "spades",
    "symbol": "ten",
    "value": "10",
  },
  Object {
    "hidden": true,
    "id": 11,
    "name": "Jack of Spades",
    "suite": "spades",
    "symbol": "jack",
    "value": "11",
  },
  Object {
    "hidden": true,
    "id": 12,
    "name": "Queen of Spades",
    "suite": "spades",
    "symbol": "queen",
    "value": "12",
  },
  Object {
    "hidden": true,
    "id": 13,
    "name": "King of Spades",
    "suite": "spades",
    "symbol": "king",
    "value": "13",
  },
  Object {
    "hidden": true,
    "id": 14,
    "name": "Ace of Clubs",
    "suite": "clubs",
    "symbol": "ace",
    "value": "1",
  },
  Object {
    "hidden": true,
    "id": 15,
    "name": "Two of Clubs",
    "suite": "clubs",
    "symbol": "two",
    "value": "2",
  },
  Object {
    "hidden": true,
    "id": 16,
    "name": "Three of Clubs",
    "suite": "clubs",
    "symbol": "three",
    "value": "3",
  },
  Object {
    "hidden": true,
    "id": 17,
    "name": "Four of Clubs",
    "suite": "clubs",
    "symbol": "four",
    "value": "4",
  },
  Object {
    "hidden": true,
    "id": 18,
    "name": "Five of Clubs",
    "suite": "clubs",
    "symbol": "five",
    "value": "5",
  },
  Object {
    "hidden": true,
    "id": 19,
    "name": "Six of Clubs",
    "suite": "clubs",
    "symbol": "six",
    "value": "6",
  },
  Object {
    "hidden": true,
    "id": 20,
    "name": "Seven of Clubs",
    "suite": "clubs",
    "symbol": "seven",
    "value": "7",
  },
  Object {
    "hidden": true,
    "id": 21,
    "name": "Eight of Clubs",
    "suite": "clubs",
    "symbol": "eight",
    "value": "8",
  },
  Object {
    "hidden": true,
    "id": 22,
    "name": "Nine of Clubs",
    "suite": "clubs",
    "symbol": "nine",
    "value": "9",
  },
  Object {
    "hidden": true,
    "id": 23,
    "name": "10 of Clubs",
    "suite": "clubs",
    "symbol": "ten",
    "value": "10",
  },
  Object {
    "hidden": true,
    "id": 24,
    "name": "Jack of Clubs",
    "suite": "clubs",
    "symbol": "jack",
    "value": "11",
  },
  Object {
    "hidden": true,
    "id": 25,
    "name": "Queen of Clubs",
    "suite": "clubs",
    "symbol": "queen",
    "value": "12",
  },
  Object {
    "hidden": true,
    "id": 26,
    "name": "King of Clubs",
    "suite": "clubs",
    "symbol": "king",
    "value": "13",
  },
  Object {
    "hidden": true,
    "id": 27,
    "name": "Ace of Diamonds",
    "suite": "diamonds",
    "symbol": "ace",
    "value": "1",
  },
  Object {
    "hidden": true,
    "id": 28,
    "name": "Two of Diamonds",
    "suite": "diamonds",
    "symbol": "two",
    "value": "2",
  },
  Object {
    "hidden": true,
    "id": 29,
    "name": "Three of Diamonds",
    "suite": "diamonds",
    "symbol": "three",
    "value": "3",
  },
  Object {
    "hidden": true,
    "id": 30,
    "name": "Four of Diamonds",
    "suite": "diamonds",
    "symbol": "four",
    "value": "4",
  },
  Object {
    "hidden": true,
    "id": 31,
    "name": "Five of Diamonds",
    "suite": "diamonds",
    "symbol": "five",
    "value": "5",
  },
  Object {
    "hidden": true,
    "id": 32,
    "name": "Six of Diamonds",
    "suite": "diamonds",
    "symbol": "six",
    "value": "6",
  },
  Object {
    "hidden": true,
    "id": 33,
    "name": "Seven of Diamonds",
    "suite": "spades",
    "symbol": "seven",
    "value": "7",
  },
  Object {
    "hidden": true,
    "id": 34,
    "name": "Eight of Diamonds",
    "suite": "diamonds",
    "symbol": "eight",
    "value": "8",
  },
  Object {
    "hidden": true,
    "id": 35,
    "name": "Nine of Diamonds",
    "suite": "diamonds",
    "symbol": "nine",
    "value": "9",
  },
  Object {
    "hidden": true,
    "id": 36,
    "name": "10 of Diamonds",
    "suite": "diamonds",
    "symbol": "ten",
    "value": "10",
  },
  Object {
    "hidden": true,
    "id": 37,
    "name": "Jack of Diamonds",
    "suite": "diamonds",
    "symbol": "jack",
    "value": "11",
  },
  Object {
    "hidden": true,
    "id": 38,
    "name": "Queen of Diamonds",
    "suite": "diamonds",
    "symbol": "queen",
    "value": "12",
  },
  Object {
    "hidden": true,
    "id": 39,
    "name": "King of Diamonds",
    "suite": "diamonds",
    "symbol": "king",
    "value": "13",
  },
  Object {
    "hidden": true,
    "id": 40,
    "name": "Ace of Hearts",
    "suite": "hearts",
    "symbol": "ace",
    "value": "1",
  },
  Object {
    "hidden": true,
    "id": 41,
    "name": "Two of Hearts",
    "suite": "hearts",
    "symbol": "two",
    "value": "2",
  },
  Object {
    "hidden": true,
    "id": 42,
    "name": "Three of Hearts",
    "suite": "hearts",
    "symbol": "three",
    "value": "3",
  },
  Object {
    "hidden": true,
    "id": 43,
    "name": "Four of Hearts",
    "suite": "hearts",
    "symbol": "four",
    "value": "4",
  },
  Object {
    "hidden": true,
    "id": 44,
    "name": "Five of Hearts",
    "suite": "hearts",
    "symbol": "five",
    "value": "5",
  },
  Object {
    "hidden": true,
    "id": 45,
    "name": "Six of Hearts",
    "suite": "hearts",
    "symbol": "six",
    "value": "6",
  },
  Object {
    "hidden": true,
    "id": 46,
    "name": "Seven of Hearts",
    "suite": "hearts",
    "symbol": "seven",
    "value": "7",
  },
  Object {
    "hidden": true,
    "id": 47,
    "name": "Eight of Hearts",
    "suite": "hearts",
    "symbol": "eight",
    "value": "8",
  },
  Object {
    "hidden": true,
    "id": 48,
    "name": "Nine of Hearts",
    "suite": "hearts",
    "symbol": "nine",
    "value": "9",
  },
  Object {
    "hidden": true,
    "id": 49,
    "name": "10 of Hearts",
    "suite": "hearts",
    "symbol": "ten",
    "value": "10",
  },
  Object {
    "hidden": true,
    "id": 50,
    "name": "Jack of Hearts",
    "suite": "hearts",
    "symbol": "jack",
    "value": "11",
  },
  Object {
    "hidden": true,
    "id": 51,
    "name": "Queen of Hearts",
    "suite": "hearts",
    "symbol": "queen",
    "value": "12",
  },
  Object {
    "hidden": true,
    "id": 52,
    "name": "King of Hearts",
    "suite": "hearts",
    "symbol": "king",
    "value": "13",
  },
  Object {
    "hidden": true,
    "id": 53,
    "name": "Joker",
    "suite": "none",
    "symbol": "joker",
    "value": 0,
  },
  Object {
    "hidden": true,
    "id": 54,
    "name": "Joker",
    "suite": "none",
    "symbol": "joker",
    "value": 0,
  },
]
`;

exports[`Cards Joker Should match snapshot 1`] = `
Object {
  "hidden": true,
  "name": "Joker",
  "suite": "none",
  "symbol": "joker",
  "value": 0,
}
`;
